src/endpoints.rs

```diff
@@ -207,3 +208,63 @@ pub async fn get_region(
             .map(|trainee| trainee.region.clone()),
     }))
 }
+
+#[derive(Serialize)]
+pub struct AttendanceResponse {
+    courses:
+        IndexMap<String, IndexMap<String, IndexMap<String, IndexMap<String, Vec<Attendance>>>>>,
+}
+
+pub async fn fecth_attendance(
+    session: Session,
+    State(server_state): State<ServerState>,
+    OriginalUri(original_uri): OriginalUri,
+) -> Result<Json<AttendanceResponse>, Error> {
+    let all_courses = &server_state.config.courses;
+    let sheets_client = sheets_client(&session, server_state.clone(), original_uri.clone()).await?;
+    let github_org = &server_state.config.github_org;
+    let octocrab = octocrab(&session, &server_state, original_uri).await?;
+    let mut courses: IndexMap<
+        String,
+        IndexMap<String, IndexMap<String, IndexMap<String, Vec<Attendance>>>>,
+    > = IndexMap::new();
+    for (course_name, course_info) in all_courses {
+        let mut course_batches: IndexMap<
+            String,
+            IndexMap<String, IndexMap<String, Vec<Attendance>>>,
+        > = IndexMap::new();
+        for batch_name in course_info.batches.keys() {
+            let course_schedule = server_state
+                .config
+                .get_course_schedule_with_register_sheet_id(course_name.clone(), &batch_name)
+                .ok_or_else(|| Error::Fatal(anyhow::anyhow!("Course not found: {course_name}")))?;
+            let mut modules: IndexMap<String, IndexMap<String, Vec<Attendance>>> = IndexMap::new();
+            let course = course_schedule
+                .with_assignments(&octocrab, github_org)
+                .await?;
+            let register_info = get_register(
+                sheets_client.clone(),
+                course.register_sheet_id.clone(),
+                course.start_date,
+                course.end_date,
+            )
+            .await?;
+
+            for (module_name, sprint_info) in register_info.modules {
+                let mut sprints: IndexMap<String, Vec<Attendance>> = IndexMap::new();
+                for (sprint_number, sprint_info) in sprint_info.attendance.into_iter().enumerate() {
+                    let mut entries: Vec<Attendance> = Vec::new();
+                    for (_index, entry) in sprint_info {
+                        entries.push(entry);
+                    }
+                    sprints.insert(format!("Sprint {}", sprint_number + 1), entries);
```

Comment:
As a rule I recommend avoiding spaces in everything wherever possible, so I'd recommend:

```suggestion
                    sprints.insert(format!("sprint-{}", sprint_number + 1), entries);
```